cert-processor/src/main/java/org/sunbird/incredible/processor/store/CloudStore.java

```diff
@@ -56,7 +56,21 @@ public void init() {
         if (StringUtils.isNotBlank(storeConfig.getType())) {
             String storageKey = storeConfig.getAccount();
             String storageSecret = storeConfig.getKey();
-            StorageConfig storageConfig = new StorageConfig(storeConfig.getType(), storageKey, storageSecret);
+            Option<String> storageEndpoint;
+            Option<String> storageRegion;
+
+            if (org.apache.commons.lang3.StringUtils.equalsIgnoreCase(storeConfig.getType(), "azure")) {
+                storageEndpoint = Option.apply("");
+                storageRegion = Option.apply("");
+            } else if (org.apache.commons.lang3.StringUtils.equalsIgnoreCase(storeConfig.getType(), "oci")){
+                storageEndpoint = Option.apply(storeConfig.getEndpoint());
+                storageRegion = Option.apply("");
+            } else {
+                storageEndpoint = scala.Option.apply("");
+                storageRegion = scala.Option.apply("");
+            }
```

Comment:
Same here as well as previous comment. We don't need storage type check, to make the code generic.

Reply:
can we take the improvements in the future releases

Reply:
removed the case statement.